Neos.ContentRepository/Migrations/Code/Version20180226141853.php

```diff
@@ -0,0 +1,109 @@
+<?php
+namespace Neos\Flow\Core\Migrations;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Exception;
+use Neos\Utility\Arrays;
+
+/**
+ * Update contentDimension Settings and convert presets to values and specialization
+ */
+class Version20180226141853 extends AbstractMigration
+{
+
+    public function getIdentifier()
+    {
+        return 'Neos.ContentRepository-20180226141853';
+    }
+
+    /**
+     * @return void
+     */
+    public function up()
+    {
+        $this->processConfiguration(
+            'Settings',
+            function (&$configuration) {
+                if ($dimensionConfigurations = Arrays::getValueByPath($configuration, 'Neos.ContentRepository.contentDimensions')) {
+                    foreach ($dimensionConfigurations as $dimensionName => $dimensionConfiguration) {
+                        $updatedConfiguration = $this->updateDimensionConfiguration($dimensionName, $dimensionConfiguration);
+                        if ($updatedConfiguration !== $dimensionConfiguration) {
+                            $configuration['Neos']['ContentRepository']['contentDimensions'][$dimensionName] = $updatedConfiguration;
+                        }
+                    }
+                }
+            },
+            true
+        );
+    }
+
+    /**
+     * @param string $dimensionConfiguration
+     * @param array $dimensionConfiguration
+     */
+    protected function updateDimensionConfiguration($dimensionName, $dimensionConfiguration)
+    {
+        //if (array_key_exists('presets', $dimensionConfiguration) && !array_key_exists('values', $dimensionConfiguration)) {
```

Comment:
remove leftover line (can be done in a followup) :)

Reply:
Is removed